src/main/scala/util/ECC.scala

```diff
@@ -176,15 +177,15 @@ class SECDEDCode extends Code
       case i if i >= w0 => i
     }
   }
-  def encode(x: UInt, poison: Bool = Bool(false)) = {
+  def encode(x: UInt, poison: Bool = false.B) = {
     // toggling two bits ensures the error is uncorrectable
     // to ensure corrected == uncorrected, we pick one redundant
     // bit from SEC (the highest); correcting it does not affect
     // corrected == uncorrected. the second toggled bit is the
     // parity bit, which also does not appear in the decoding
     val toggle_lo = Cat(poison.asUInt, poison.asUInt)
     val toggle_hi = toggle_lo << (sec.width(x.getWidth)-1)
-    par.encode(sec.encode(x)) ^ toggle_hi
+    par.encode(sec.encode(x)) ^ toggle_hi.asUInt
```

Comment:
```suggestion
    par.encode(sec.encode(x)) ^ toggle_hi
```
I don't think here we need `asUInt`